lib/fika/compiler/type_checker/parallel_type_checker.ex

```diff
@@ -52,21 +53,20 @@ defmodule Fika.Compiler.TypeChecker.ParallelTypeChecker do
   def handle_continue(:start, state) do
     pid = self()
 
-    state.unchecked_functions
-    |> Task.async_stream(
-      fn {signature, function} ->
-        result = TypeChecker.check(function, %{type_checker_pid: pid})
-        __MODULE__.post_result(pid, signature, result)
-      end,
-      max_concurrency: Enum.count(state.unchecked_functions)
-    )
-    |> Enum.each(fn
-      {:ok, _} ->
-        :ok
+    Enum.each(state.unchecked_functions, fn {signature, function} ->
+      # We need to use this approach instead of Task.async_stream due to
```

Comment:
@emilsoman I had to roll this back because some tests were hanging. I've added this comment to make it clear why we take this approach

Reply:
Sure, we can just say the tests were hanging and which ones as part of the comment to make it clearer